File: src/src_data/fmg/data/view/draw/EShowElement.java
package fmg.data.view.draw;

public enum EShowElement {
	eCaption,
	eMenu,
	eToolbar,
	eStatusbar;

	public String getDescription() {
		switch (this) {
		case eCaption  : return "Show caption";
		case eMenu     : return "Show menu";
		case eToolbar  : return "Show toolbar";
		case eStatusbar: return "Show status bar";
		}
		throw new RuntimeException("Invalid "+this.toString());
	}
}

File: src/src_core/fmg/core/types/draw/EShowElement.cs
using System;

namespace fmg.core.types.draw {

    public enum EShowElement {
        eCaption,
        eMenu,
        eToolbar,
        eStatusbar
    }

    public static class EShowElementExt {
        public static string getDescription(this EShowElement self) {
            switch (self) {
            case EShowElement.eCaption: return "Show caption";
            case EShowElement.eMenu: return "Show menu";
            case EShowElement.eToolbar: return "Show toolbar";
            case EShowElement.eStatusbar: return "Show status bar";
            }
            throw new Exception("Invalid " + self.ToString());
        }
    }

}
